cattle/plugins/docker/storage.py

```diff
@@ -112,7 +111,4 @@ def _do_volume_remove(self, volume, storage_pool, progress):
                     shutil.rmtree(path)
 
     def _path_to_volume(self, volume):
-        host_path = volume.uri.replace('file://', '')
-        mounted_path = re.sub('^.*?/var/lib/docker', '/host/var/lib/docker',
-                              host_path)
-        return mounted_path
+        return volume.uri.replace('file://', '')
```

Comment:
Crap... No.   This is going to cause problems.  Let me think more about this.  Both my change and the existing code will fail on Ubuntu snappy because docker is not at /var/lib/docker.   Through bind mount magic I take the random path on the host and bind mount it to var/lib/docker in the agent container.

The best hack I can think of is to look for the vfs directory and orient yourself from there....